packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -163,8 +164,7 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
 
     // If the selectedKey changed, update the input value.
     // Do nothing if both inputValue and selectedKey are controlled.
-    // In this case, it's the user's responsibility to update inputValue in onSelectionChange. In addition, we preserve the defaultInputValue
-    // on initial render, even if it doesn't match the selected item's text.
```

Comment:
last bit of this comment was outdated since the reseting is done in the if statement immediately above now